src/jvp.jl

```diff
@@ -0,0 +1,196 @@
+mutable struct JVPCache{X1, FX1, FDType}
+    x1  :: X1
+    fx1 :: FX1
+end
+
+"""
+    FiniteDiff.JVPCache(
+        x,
+        fdtype     :: Type{T1} = Val{:forward})
+
+Allocating Cache Constructor.
+"""
+function JVPCache(
+    x,
+    fdtype::Union{Val{FD},Type{FD}} = Val(:forward)) where {FD}
+    fdtype isa Type && (fdtype = fdtype())
+    JVPCache{typeof(x), typeof(x), fdtype}(copy(x), copy(x))
+end
+
+"""
+    FiniteDiff.JVPCache(
+        x,
+        fx1,
+        fdtype     :: Type{T1} = Val{:forward},
+
+Non-Allocating Cache Constructor.
```

Comment:
well it does make a copy of `x`

Reply:
yes the non-allocating cache is expected to just use the arrays as given 😅